Client/Frontend/Browser/BrowserPrompts.swift

```diff
@@ -15,8 +15,31 @@ import Shared
 /// need to invoke the completionHandler passed to us from the WKWebView delegate or else
 /// a runtime exception is thrown.
 class JSPromptAlertController: UIAlertController {
-    var alertInfo: JSAlertInfo?
-
+    
+    var info: JSAlertInfo!
+    var showsCancel: Bool = false
+    
+    class func newController(title: String?, message: String?, preferredStyle: UIAlertController.Style = .alert, info: JSAlertInfo, showCancel: Bool = true) -> JSPromptAlertController {
```

Comment:
Thats not possible since, `init(title: title, message: message, preferredStyle: preferredStyle)` is a convenience initalizer itself. And the only init we can override is ` override init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: Bundle?)` which if used requires all params  to be nil and the super init call to be for the convenience initalizer

Reply:
Convenience init, can call a convenience init.

Reply:
Yeah just realised, thanks!